bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/handler/DaikinMadokaHandler.java

```diff
@@ -0,0 +1,761 @@
+            logger.debug("[{}] Bridge is null. Exiting.", super.thing.getUID().getId());
+            return;
+        }
+
+        this.commandExecutor = Executors
+                .newSingleThreadExecutor(new NamedThreadFactory(thing.getUID().getAsString(), true));
+
+        this.refreshJob = scheduler.scheduleWithFixedDelay(() -> {
+            // It is useless to refresh version all the time ! Just once.
+            if (this.madokaSettings.getCommunicationControllerVersion() == null
+                    || this.madokaSettings.getRemoteControllerVersion() == null) {
+                submitCommand(new GetVersionCommand());
+            }
+            submitCommand(new GetIndoorOutoorTemperatures());
+            submitCommand(new GetOperationmodeCommand());
+            submitCommand(new GetPowerstateCommand()); // always keep the "GetPowerState" aftern the "GetOperationMode"
+            submitCommand(new GetSetpointCommand());
+            submitCommand(new GetFanspeedCommand());
+        }, 10, this.config.refreshInterval, TimeUnit.SECONDS);
+
+    }
+
+    @Override
+    public void dispose() {
+
+        logger.debug("[{}] dispose()", super.thing.getUID().getId());
+
+        dispose(refreshJob);
+        dispose(commandExecutor);
+        dispose(currentCommand);
+
+        // Unsubscribe to characteristic notifications
+        if (this.device != null) {
+            BluetoothCharacteristic charNotif = this.device
+                    .getCharacteristic(DaikinMadokaBindingConstants.CHAR_NOTIF_UUID);
+
+            if (charNotif != null) {
+                @NonNull
+                BluetoothCharacteristic c = charNotif;
+                this.device.disableNotifications(c);
+            }
+        }
+
+        // Disconnect from the device
+        if (this.device != null) {
+            this.device.disconnect();
+        }
+
+        super.dispose();
+    }
+
+    private static void dispose(@Nullable ExecutorService executor) {
+        if (executor != null) {
+            executor.shutdownNow();
+        }
+    }
+
+    private static void dispose(@Nullable ScheduledFuture<?> future) {
+        if (future != null) {
+            future.cancel(true);
+        }
+    }
+
+    private static void dispose(@Nullable BRC1HCommand command) {
+        if (command != null) {
+            // even if it already completed it doesn't really matter.
+            // on the off chance that the commandExecutor is waiting on the command, we can wake it up and cause it to
+            // terminate
+            command.setState(BRC1HCommand.State.FAILED);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("[{}] Channel: {}, Command: {}", super.thing.getUID().getId(), channelUID, command);
+
+        if (command instanceof RefreshType) {
+            // The refresh commands are not supported in query mode.
+            // The binding will notify updates on channels
```

Comment:
What do you mean by *query mode*?
Why can't you use your getter commands to handle the refresh?

Reply:
It was just done on purpose - channels are only refreshed when the bindings decides it - ie. with the refresh timer. It was to avoid too many refreshes coming from elsewhere. The BRC1H does not handle very well too many queries, and when it is stuck, you have to completely power off your entire AC installation (from the switchboard) which is not super convienient.